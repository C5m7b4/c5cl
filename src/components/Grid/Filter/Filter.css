.mikto-table-filter-title {
  margin-left: 8px;
}

.mikto-table-filter-item {
  display: grid;
  grid-template-columns: 20px 1fr;
}

.mikto-table-filter-header {
  border-bottom: 3px solid black;
  margin-top: 5px;
  margin-bottom: 5px;
  padding: 4px;
}
