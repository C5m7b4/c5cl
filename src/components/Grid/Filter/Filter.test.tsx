import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Filter } from './Filter';

const header = {
  columnName: 'id',
  title: 'ID',
  visible: true,
  style: {
    textAlign: 'center',
  },
};

const testData = [
  {
    id: 1,
    storeName: 'IGA 001',
    storeNumber: '001',
    termCount: 3,
  },
  {
    id: 2,
    storeName: 'IGA 002',
    storeNumber: '002',
    termCount: 4,
  },
  {
    id: 3,
    storeName: 'IGA 003',
    storeNumber: '003',
    termCount: 6,
  },
];

describe('Filter', () => {
  test('should render correctly', () => {
    const frag = render(
      // @ts-ignore
      <Filter header={header} open={true} divId="tf" data={testData} />
    );

    expect(frag).toMatchSnapshot();
  });
  test('should show open modal', () => {
    document.body.innerHTML = '<div id="tf"></div>';

    const frag = render(
      <Filter
        // @ts-ignore
        header={header}
        open={true}
        divId="tf"
        data={testData}
        availableFilters={[]}
      />
    );
    expect(frag).toMatchSnapshot();
  });
  test('should detect if an item is already checked', () => {
    document.body.innerHTML = '<div id="tf"></div>';

    const frag = render(
      <Filter
        // @ts-ignore
        header={header}
        open={true}
        divId="tf"
        data={testData}
        availableFilters={['ID']}
      />
    );
    expect(frag).toMatchSnapshot();
  });
  test('should detect checkbox click', () => {
    document.body.innerHTML = '<div id="tf"></div>';

    const testFn = jest.fn();

    const frag = render(
      <Filter
        // @ts-ignore
        header={header}
        open={true}
        divId="tf"
        data={testData}
        availableFilters={['ID']}
        filterItemClicked={testFn}
      />
    );
    const check: HTMLInputElement = screen.getByTestId('mikto-table-filter-0');
    userEvent.click(check);
  });
});
