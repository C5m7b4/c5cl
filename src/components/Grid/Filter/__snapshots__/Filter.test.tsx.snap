// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should detect if an item is already checked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="tf"
    >
      <div>
        <div
          class="mikto-table-filter-header"
        >
          Filters
        </div>
        <div
          class="mikto-table-filter-item"
        >
          <input
            class="mikto-table-filter-checkbox"
            data-testid="mikto-table-filter-0"
            type="checkbox"
          />
          <span
            class="mikto-table-filter-title"
          />
        </div>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Filter should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Filter should show open modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="tf"
    >
      <div>
        <div
          class="mikto-table-filter-header"
        >
          Filters
        </div>
        <div
          class="mikto-table-filter-item"
        >
          <input
            class="mikto-table-filter-checkbox"
            data-testid="mikto-table-filter-0"
            type="checkbox"
          />
          <span
            class="mikto-table-filter-title"
          />
        </div>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
